### ISSUE TITLE
Add an API based on websocket

### ISSUE THREAD

**Issue by MainRo** [MainRo/deepspeech-server#4]
# Objective

Implement an API based on websocket. Compared to the http API, this will improve latency because audio data will be send on the fly as it is available. 

If possible, the API should follow the reactive streams convention, where a stream is a request to do a STT inference (from client to server), and another stream carries the answer (from server to client):

- Request stream creation is done with one message.
- several items a sent on the request stream to carry the audio data (split by chunks).
- The request stream completes once all audio data is transfered or fails in case of error. 

When the request stream completes, a STT inference is done, and the anser is sent to a response stream. So there is one request stream per request and a common response stream.
 
# Specification

# Alternative solutions

**Comment by cogmeta** [comment]
Let us know if we can help with websocket/socketio version?

**Comment by MainRo** [comment]
I did not work on this part yet. We can start by defining the API here. We can share ideas here and I will merge them in the first comment. Once this is done I think that I can do the implementation on the server rather quickly. Moreover, implementing a javascript client library embeddable in a browser would be nice to ease integration in a web app. Tell me on what part(s) you want to contribute.

Originally I though about using a code generator that I started to implement (https://github.com/rxtender/rxtender), on top of a websocket connexion (using aiohttp). However this generator is not ready yet. So it is better to implement the audio transmission logic by hand.

Using socketio may be an option instead of direct websocket usage. I see that there is a python package that relies on aiohttp (python-socketio), so it should fit well with the current code.
